projects/core/src/cart/store/cart-state.ts

```diff
@@ -17,4 +19,5 @@ export interface CartState {
   entries: { [code: string]: OrderEntry };
   refresh: boolean;
   cartMergeComplete: boolean;
+  appliedVouchers: Voucher[];
```

Comment:
Since you get vouchers from cart.vouchers, I think this state maybe unnecessary and the same as selector & reducer.

Reply:
remove the state/reducer/selector from cart since the voucher can be get from cart directly.